sdk/instruction/Cargo.toml

```diff
@@ -0,0 +1,57 @@
+[package]
+name = "solana-instruction"
+description = "Types for directing the execution of Solana programs."
+documentation = "https://docs.rs/solana-instruction"
+version = { workspace = true }
+authors = { workspace = true }
+repository = { workspace = true }
+homepage = { workspace = true }
+license = { workspace = true }
+edition = { workspace = true }
+
+[dependencies]
+bincode = { workspace = true, optional = true }
+borsh = { workspace = true, optional = true }
+num-traits = { workspace = true }
+serde = { workspace = true, optional = true }
+serde_derive = { workspace = true, optional = true }
+solana-frozen-abi = { workspace = true, optional = true }
+solana-frozen-abi-macro = { workspace = true, optional = true }
+solana-pubkey = { workspace = true, default-features = false }
+solana-sanitize = { workspace = true }
+solana-short-vec = { workspace = true, optional = true }
+
+[target.'cfg(target_os = "solana")'.dependencies]
+solana-define-syscall = { workspace = true }
+
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+getrandom = { workspace = true, features = ["js", "wasm-bindgen"] }
+js-sys = { workspace = true }
+wasm-bindgen = { workspace = true }
+
+[dev-dependencies]
+solana-instruction = { path = ".", features = ["borsh"] }
+
+[build-dependencies]
+rustc_version = { workspace = true, optional = true }
+
+[features]
+bincode = ["dep:bincode", "dep:serde"]
+borsh = ["dep:borsh"]
```

Comment:
No because the only the borsh functionality here is this associated function which does not require any borsh stuff from `Pubkey`:

```rust
    pub fn new_with_borsh<T: BorshSerialize>(
        program_id: Pubkey,
        data: &T,
        accounts: Vec<AccountMeta>,
    ) -> Self {
        let data = borsh::to_vec(data).unwrap();
        Self {
            program_id,
            accounts,
            data,
        }
    }
```